service/controller/cluster_resource_set.go

```diff
@@ -558,6 +559,19 @@ func newClusterResourceSet(config clusterResourceSetConfig) (*controller.Resourc
 		}
 	}
 
+	var ensureCPCRsResource resource.Interface
```

Comment:
I get the naming now but `CPCR` is a really confusing acronym. It took me a while to realise that is `CP` and then `CR` and not some `CPC` `R` or whatever.

Is there a way to make it more readable?

Reply:
Sure, you suggest something and I rename it. I don't mind.

Reply:
Following the `vpc` naming would help: `cpVPCResource`

`cpCRsResource` at least has different capitalization.

Reply:
I think that does not really improve the situation. Note we have also things like `CPCIDR` and `VPCID`.

Reply:
Since there aren't any great naming options here, how about a running glossary of acronyms or a quick index of what each resource does?

Reply:
Like `README.md` in the `service/controller/resource` package? I like that somehow. There is only the issue that the code should tell the story to a certain extend and I would not like to improve documentation where actually the code should be improved.

Reply:
Yes, I prefer self documenting code. When it's not possible, a single line comment describing what the variable/resource is would seem to be a good compromise.

Reply:
When you think it is worth it, go for it. I do not need it to be honest. @lachlanh may have an opinion on this as well.